Board.java

```diff
@@ -0,0 +1,634 @@
+            SingleBoard = new Board();
+        }
+        return SingleBoard;
+    }
+
+    public Tile[][] getTiles() {
+        Tile[][] tilesBoard = new Tile[BOARD_SIZE][BOARD_SIZE];
+
+        for (int i = 0; i < board[0].length; i++) {
+            for (int j = 0; j < board.length; j++) {
+                tilesBoard[i][j] = board[i][j].getTile();
+            }
+        }
+
+        return tilesBoard;
+    }
+
+    public boolean boardLegal(Word word) {
+        return isWordFitInBoard(word) && isWordLayOnOrNextToOther(word) && !isReplacingATile(word);
+    }
+
+    private boolean isWordFitInBoard(Word word) {
+
+        int row = word.getRow();
+        int col = word.getCol();
+
+        if(row < 0 || row > BOARD_SIZE - 1 || col < 0 || col > BOARD_SIZE - 1) {
+            return false;
+        }
+
+        int wordLength = word.getTiles().length;
+
+        if(word.isVertical()){
+            return row + wordLength <= BOARD_SIZE - 1;
+        }
+        return col + wordLength <= BOARD_SIZE - 1;
+    }
+
+    private boolean isReplacingATile(Word word) {
+        int row = word.getRow();
+        int col = word.getCol();
+        Tile[] wordTiles = word.getTiles();
+
+        if(word.isVertical()) {
+            for (int i = row; i < row + wordTiles.length; i++) {
+                if (wordTiles[i - row] != null && board[i][col].isOccupied()) {
+                        return true;
+                }
+            }
+        }
+        else {
+            for (int i = col; i < col + wordTiles.length; i++) {
+                if (wordTiles[i - col] != null && board[row][i].isOccupied()) {
+                        return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private boolean isWordLayOnOrNextToOther(Word word) {
+        int row = word.getRow();
+        int col = word.getCol();
+        int wordLength = word.getTiles().length;
+        int centerIndex = BOARD_SIZE / 2;
+
+        // Very much not efficient - but for now
+        boolean isFirstTurn = true;
+        for (int i = 0; i < board[0].length; i++) {
+            for (int j = 0; j < board.length; j++) {
+                if(board[i][j].isOccupied()) {
+                    isFirstTurn = false;
+                    break;
+                }
+            }
+        }
+
+        if(isFirstTurn) {
+            if(word.isVertical()) {
+                return row <= centerIndex && centerIndex <= row + wordLength && col == centerIndex;
+            } else {
+                return col <= centerIndex && centerIndex <= col + wordLength && row == centerIndex;
+            }
+        }
+
+        return isValidPlacement(word);
+    }
+
+    private boolean isValidPlacement(Word word) {
+        int row = word.getRow();
+        int col = word.getCol();
+        Tile[] wordTiles = word.getTiles();
+
+
+        if (word.isVertical()) {
+            // Check if the first or last letter of the word has a vertical neighbor
```

Comment:
personally i prefer to write a function instead of comment. for example - 
if (checkIfTheFirstOrLastLetterHasAVerticalNeighbor())

Reply:
thanks for the tip, in this case i use those functions for more things so i dont think that here it will be a good fit